OpenSim/Simulation/SimbodyEngine/Test/testJoints.cpp

```diff
@@ -2226,6 +2231,42 @@ void testAutomaticJointReversal()
     ASSERT_EQUAL(accErr, 0.0, sqrt(integ_accuracy));
 }
 
+void testUserJointReversal()
+{
+    using namespace OpenSim;
+
+    cout << "\n==========================================================="
+         << "\n Test joint reversal set by user in old model file"
+         << "\n==========================================================="
+         << endl;
+
+    // Open model.
+    auto model = Model("double_pendulum_testReverse.osim");
+    //model.finalizeFromProperties();
+    model.initSystem();
```

Comment:
Sorry that comment needs to be updated, but it doesn't have any bearing on the current issue as far as I can tell. 
Does `finalizeFromProperties()` itself cause the throw or only when you try to access the Socket afterward? `finalizeFromProperties()` does not invoke `finalizeConnections()`. If you need to resolve and access connection information you need to call `finalizeConnections()`, which `initSystem()` does as well.